src/reactviews/pages/ContainerDeployment/stepCard.tsx

```diff
@@ -0,0 +1,132 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Spinner, Card, Button, makeStyles, tokens } from "@fluentui/react-components";
+import {
+    Checkmark20Regular,
+    ChevronDown20Regular,
+    ChevronUp20Regular,
+    Dismiss20Regular,
+} from "@fluentui/react-icons";
+import { useContext, useState } from "react";
+import { ContainerDeploymentContext } from "./containerDeploymentStateProvider";
+import { ApiStatus } from "../../../sharedInterfaces/webview";
+import { DockerStep } from "../../../sharedInterfaces/containerDeploymentInterfaces";
+import { locConstants } from "../../common/locConstants";
+
+const useStyles = makeStyles({
+    outerDiv: {
+        height: "fit-content",
+        width: "500px",
+        position: "relative",
+        overflow: "auto",
+    },
+    spinnerDiv: {
+        display: "flex",
+        justifyContent: "center",
+        alignItems: "center",
+        flexDirection: "column",
+        padding: "20px",
+    },
+    leftHeader: {
+        display: "flex",
+        alignItems: "center",
+        gap: "8px",
+    },
+    separatorDiv: {
+        position: "absolute",
+        left: 0,
+        top: 0,
+        bottom: 0,
+        width: "4px",
+        background: tokens.colorNeutralStroke2,
+    },
+    header: {
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "space-between",
+        padding: "4px",
+    },
+    topSpace: {
+        marginTop: "8px",
+    },
+});
+
+interface StepCardProps {
+    step: DockerStep;
+}
+
+export const StepCard: React.FC<StepCardProps> = ({ step }) => {
+    const classes = useStyles();
+    const state = useContext(ContainerDeploymentContext);
+    const [expanded, setExpanded] = useState(false);
+    const [showFullErrorText, setShowFullErrorText] = useState(false);
+
+    // If this passes, container deployment state is guaranteed
+    // to be defined, so we can reference it as non-null
+    if (!state) {
+        return undefined;
+    }
+
+    const getStatusIcon = () => {
+        if (step.loadState === ApiStatus.Loaded) {

```

Comment:
How about adding a 'Not Started' State. The UX gets a little confusing with all the steps in loading state at once.